### PR TITLE
Formatting Number Fix

### PR DESCRIPTION
**Pull request by knightsean00**
https://issues.redhat.com/browse/AA-640

### CODE REVIEW

File: src/components/Functions/axisFormat.ts [hunk 1]
```diff
@@ -11,10 +11,25 @@ export const formatDateAsDayMonth = (i: string): string => {
     return `${parts[1]}/${parts[2]}`;
 }
 
-export const formatNumberAsK = (n: string | number = 0): string =>
-    (Math.abs(+n) > 1000)
-        ? `${(+n / 1000).toFixed(1)}k`
-        : `${(+n).toFixed(0)}`;
+export const formatNumberAsK = (n: string | number = 0): string => {
+    if (+n === 0) {
```
Comment (brumik) [inline src/components/Functions/axisFormat.ts:15]: The preferred method to convert something to string is  `{stringVar}`. Can we use it here too (and bellow).

I prefer having a single return statement without `{` and `}` inlined.

**Review by brumik — CHANGES_REQUESTED** [review]
Can you please make the demo data with a positive cumulative value too? Having only negative ones does not really reflect the expected use case.

Btw: was this feature requested somewhere?

**Comment by knightsean00** [comment]
> Can you please make the demo data with a positive cumulative value too? Having only negative ones does not really reflect the expected use case.
> 
> Btw: was this feature requested somewhere?

File: src/components/Functions/axisFormat.ts [hunk 2]
```diff
@@ -11,10 +11,21 @@ export const formatDateAsDayMonth = (i: string): string => {
...
+    if (Math.abs(+n) > 1000000) {
+        divisor = 1000000;
+        suffix = 'M';
+    } else if (Math.abs(+n) > 1000) {
+        divisor = 1000;
+        suffix = 'k';
+    }
+
+    const output = +n / divisor;
+    const remainLen = 3 - `${output.toFixed(0)}`.length + (output > 0 ? 0 : 1);
```
Comment (brumik) [inline src/components/Functions/axisFormat.ts:26]: I now understand the number `3` at the start of the row, can we move it to a constant and name it appropriately so it is easier to read?

**Comment by brumik** [comment]
I made the function to return to 1 decimal point the digits, since in cases where the range is going by for example 500 you would only get 0, 500, 1k, 1k, 2k which is not ideal. If we do millions then it is the same in the higher range where you can possibly get after each other in the label the same number. As I seen the issue they don't think about it. I would suggest to have 4 or 5 digists where the last or last two digits are decimal spaces so we get with much smaller chance the same labels for two different spots.

**Comment by knightsean00** [comment]
> I made the function to return to 1 decimal point the digits, since in cases where the range is going by for example 500 you would only get 0, 500, 1k, 1k, 2k which is not ideal. 

In this case, you would get 0, 500, 1k, 1.5k, and 2k which I think works fine?

> I would suggest to have 4 or 5 digists where the last or last two digits are decimal spaces so we get with much smaller chance the same labels for two different spots.

That should be fine because if our scale is large, then three sig figs are more than enough to differentiate as long as we have a large range (e.g. I don't think we would ever have a graph where the tick marks would just be small variations of 2M)

**Review by brumik — APPROVED** [review]
Okey, I see. If there will be a graph with a small variability over big number we can just revisit this function then :)

**Pull request merged**